delfin/ssl_utils.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import ssl
+import requests
+from oslo_config import cfg
+from oslo_log import log
+from urllib3 import PoolManager
+
+from delfin import exception
+
+LOG = log.getLogger(__name__)
+CONF = cfg.CONF
+FILE = 'configs.json'
+
+
+def verify_ca_path(verify):
+    """
+    Checking the ca_path exists
+    """
+    if isinstance(verify, str):
+        if not os.path.exists(verify):
+            LOG.error("Directory {0} could not be found.".format(
+                verify))
+            raise exception.InvalidCAPath(verify)
+
+
+def get_configs():
+    bool_val = {'True': True, 'False': False}
+    verify = CONF.security.verify
+    return bool_val.get(verify, verify)
+
+
+class HostNameIgnoreAdapter(requests.adapters.HTTPAdapter):
+    def cert_verify(self, conn, url, verify, cert):
+        conn.assert_hostname = False
+        return super(HostNameIgnoreAdapter, self).cert_verify(
+            conn, url, verify, cert)
+
+    def init_poolmanager(self, connections, maxsize, block=False,
+                         **pool_kwargs):
+        self._pool_connections = connections
+        self._pool_maxsize = maxsize
+        self._pool_block = block
+        self.poolmanager = PoolManager(num_pools=connections, maxsize=maxsize,
+                                       block=block, strict=True,
+                                       ssl_version=ssl.PROTOCOL_TLSv1_2,
```

Comment:
PROTOCOL_TLSv1_2 should be configurable, some system doesn't support TLSv1_2, u can create __init__ to do it. 

```
def __init__(ssl_version=ssl.PROTOCOL_TLSv1_2):
    self.ssl_vresion=ssl_version
```

then
```
PoolManager(num_pools=connections, maxsize=maxsize,
                                       block=block, strict=True,
                                       ssl_version=self.ssl_vresion,
 
```

Reply:
This did not work currently